src/test/java/org/broadinstitute/barclay/help/DefaultDocWorkUnitHandlerUnitTest.java

```diff
@@ -80,6 +81,19 @@ public void testGetGroupSummaryForWorkUnit(final Class<?> docWorkUnitClazz, fina
         Assert.assertEquals(handler.getGroupSummaryForWorkUnit(mockWorkUnit), expectedGroupSummary);
     }
 
+    @DocumentedFeature
+    @CommandLineProgramProperties(summary = "CLP without no-args constructor", oneLineSummary = "CLP without no-args constructor", programGroup = TestProgramGroup.class)
+    private final class ClpWithoutNoArgsConstructor {
+        public ClpWithoutNoArgsConstructor(final int arg) { }
+    }
+
+    @Test(expectedExceptions = RuntimeException.class)
+    public void testProcessWorkUnitWithoutNoArgsConstructorClpThrows() {
+        final DefaultDocWorkUnitHandler handler = getDefaultWorkUnitHandlerInstance();
+        final DocWorkUnit workUnit = createMockWorkUnit(handler, ClpWithoutNoArgsConstructor.class, "", Collections.emptyMap());
+        handler.processWorkUnit(workUnit, Collections.emptyList(), Collections.emptyList());
+    }
```

Comment:
Nice! We might as well add the converse as a unit test as well; a DocumentedFeature that doesn't have a no-arg constructor and doesn't have a CommandLineProgram annotation. Then just validate that the resulting workUnit property(s).

Reply:
The current mocking framework is not able to process the work unit because it is not mocking the `ClassDoc.fields` and thus it throws a NPE, and even if it is mocking it, the codepaths for the `FieldDocs` will change in #105.

I think that testing the rest of the `processWorkUnit()` should be added in a different PR, because it requires much more work such as testing the different codepaths to populate different properties. I added a disabled test to show the issue...

Reply:
Yeah, I see your point. I created https://github.com/broadinstitute/barclay/issues/107 for that. In the meantime lets not hold up these bug fix PRs since they're covered by integration tests.